298 add missing commands for the vested staking

Vitomir2 opened the pull request:
# Description

- update the bindings to include the stakeWithVesting and getValidator;
- add new flags - vesting and vesting-period, and verify the period to e in the valid period;
- create a new createStakeTransaction function in order to make the code more readable and cleaner;
- optimize the code in the logs loop;
- make the amount flag mandatory;
- mark flags mutually exclusive with the delegate flag;
- make a MaxVestingPeriod constant in the sidechain helper;
- fix the conditions in the commission command event handling;
- rename stakeManager -> hydraStaking in the register_validator to avoid confusion;
- create ValidatorStatus enum and GetStatus function in the validator package;
- create a new getValidatorInfo helper function to get the data that I need;
- update docs;

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [x] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [ ] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [x] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

Vitomir2 committed:
add the stake vesting functionality

update the bindings to include the stakeWithVesting;
add new flags - vesting and vesting-period, and verify the period to e in the valid period;
create a new createStakeTransaction function in order to make the code more readable and cleaner;
optimize the code in the logs loop;
make the amount flag mandatory;
mark flags mutually exclusive with the delegate flag;
make MaxVestingPeriod constant in the sidechain helper;

Vitomir2 committed:
updates

fix the conditions in the commission command event handling;
rename stakeManager -> hydraStaking in the register_validator to avoid confusion;
update the bindings to contain the getValidator function from the HydraChain contract;
create a new getValidatorInfo helper function to get the data that I need;
update docs;

Vitomir2 committed:
cleanup the code from comments

Vitomir2 committed:
move ValidatorStatus to validator package and update docs

Vitomir2 committed:
cleanup the codebase

Vitomir2 committed:
add in weeks

Vitomir2 committed:
move the ValidatorStatus from the validator_set to validator_metadata

R-Santev commented on README.md:
```diff
@@ -185,6 +183,20 @@ hydra hydragon stake --data-dir ./node-secrets --self true --amount 150000000000
+
+Hydra Chain allows you to open a vested position, where your funds are locked for a chosen period between 1 and 52 weeks (1 year). In return, you receive a loyalty bonus on the APR. The longer the vesting duration, the higher the bonus.
+
+You can unstake your funds prematurely by paying a penalty fee, which is calculated at 0.5% per remaining week of the lockup period. Consequently, the closer the position is to maturity, the lower the penalty fee, while the further away, the higher the cost. Additionally, any rewards that haven`t completed vesting will also be burned in the process. For more details on this mechanism, refer to th ...
```
```suggestion
You can unstake your funds prematurely by paying a penalty fee, which is calculated at 0.5% per remaining week of the lockup period. Consequently, the closer the position is to maturity, the lower the penalty fee, while the further away, the higher the cost. Additionally, any rewards distributed to you in the vesting period will also be burned in the process. For more details on this mechanism, refer to the Whitepaper.
```

R-Santev commented on command/sidechain/staking/params.go:
```diff
@@ -18,45 +20,74 @@ type stakeParams struct {
 	jsonRPC            string
 	amount             string
 	self               bool
+	vesting            bool
```
no need of two flags

R-Santev commented on consensus/polybft/validator/validator_metadata.go:
```diff
@@ -32,6 +32,30 @@ type ValidatorMetadata struct {
 	IsActive    bool
 }
 
+// Define the ValidatorStatus enum
```
better move this to the commands section

Vitomir2 committed:
Update README.md

Co-authored-by: Rosen Santev <77731162+R-Santev@users.noreply.github.com>

Vitomir2 committed:
optimization

move the ValidatorStatus and the GetStatus inside the sidechain helper;
remove the vesting flag and use only the vesting-period;
update the readme;
